org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/commands/ProjectCommand.java

```diff
@@ -175,11 +175,11 @@ public static Map<String, Object> getProjectSettings(String uri, List<String> se
 							continue;
 						} else if (entryKind == IClasspathEntry.CPE_LIBRARY) {
 							if (!path.toFile().exists()) {
-								// the case when lib path is relative
-								path = project.getFile(path.makeRelativeTo(project.getFullPath())).getLocation();
-							}
-							if (!path.toFile().exists()) {
-								continue;
+								// check if it's a project based full path
+								IPath filePath = project.getFile(path.makeRelativeTo(project.getFullPath())).getLocation();
+								if (filePath != null && filePath.toFile().exists()) {
```

Comment:
Before the change, we were skipping the `path` (continue) if it still didn't exist after the attempt to resolve it. Now if it's `null` or doesn't exist, we're leaving it relative, and progressing further in the loop where it's added to the returned map. Do we want that ?

Reply:
> Do we want that? Is it up to the client to deal with it at this point ?

Yes, my thought is that then user has a chance to do things like removing the nonexistent library.